app/services/hyrax/file_set_fixity_check_service.rb

```diff
@@ -1,9 +1,45 @@
 module Hyrax
+  # This class runs fixity checks on a FileSet, potentially on multiple
+  # files each with multiple versions in the FileSet.
+  #
+  # The FixityCheck itself is performed by FixityCheckJob, which
+  # just uses the fedora service to ask for fixity verification.
+  # The outcome will be some created ChecksumAuditLog (ActiveRecord)
+  # objects, recording the checks and their results.
+  #
+  # By default this runs the checks async using ActiveJob, so
+  # returns no useful info -- the checks are still going on. Use
+  # FixityStatusService if you'd like a human-readable status based on
+  # latest recorded checks, or ChecksumAuditLog.latest_for_fileset_id
+  # if you'd like the the machine-readable checks.
+  #
+  # But if you initialize with `async_jobs: false`, checks will be done
+  # blocking in foreground, and you can get back the ChecksumAuditLog
+  # records created.
+  #
+  # It will only run fixity checks if there are not recent
+  # ChecksumAuditLogs on record. "recent" is defined by
+  # `max_days_between_fixity_checks` arg, which defaults to config'd
+  # `Hyrax.config.max_days_between_fixity_checks`
   class FileSetFixityCheckService
-    attr_reader :file_set, :id
+    attr_reader :file_set, :id, :latest_version_only,
+                :async_jobs, :max_days_between_fixity_checks
 
     # @param file_set [ActiveFedora::Base, String] file_set
-    def initialize(file_set)
+    # @param async_jobs [Boolean] Run actual fixity checks in background. Default true.
+    # @param max_days_between_fixity_checks [int] if an exisitng fixity check is
+    #   recorded within this window, no new one will be created. Default
+    #   `Hyrax.config.max_days_between_fixity_checks`. Set to -1 to force
+    #    check.
+    # @param latest_version_only [Booelan]. Check only latest version instead of all
+    #   versions. Default false.
+    def initialize(file_set,
```

Comment:
I assume you mean 'signature', not 'erasure'?  See above on my current attitude toward deprecation.  The previous signature was just too easy to be mis-used (and was being mis-used in existing code), it needed a new signature that was clearer on semantics and intent. 

I guess one could leave the signature the same (subject to misuse) for sake of backwards compat, but I think this would be unfortunate. Or one could add deprecation with previous signature working here and in other places, with significant increase in complexity of code. 

Originally I wanted to maintain backwards compat everywhere, but the more I got into it, the more I arrived at thinking this stuff had to be more drastically altered to have a transparent and clear API.